addons/web/static/tests/views/list_tests.js

```diff
@@ -1501,6 +1501,173 @@ QUnit.module('Views', {
         list.destroy();
     });
 
+    QUnit.test('editable list: list view in an initially unselected notebook page', async function (assert) {
+        assert.expect(9);
+
+        this.data.foo.records = [{ id: 1, o2m: [1] }];
+        this.data.bar = {
+            fields: {
+                titi: { string: "Small char", type: "char", sortable: true },
+                grosminet: { string: "Beeg char", type: "char", sortable: true },
+            },
+            records: [
+                {
+                    id: 1,
+                    titi: "Tiny text",
+                    grosminet:
+                        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. ' +
+                        'Ut at nisi congue, facilisis neque nec, pulvinar nunc. ' +
+                        'Vivamus ac lectus velit.',
+                },
+            ],
+        };
+        var form = await createView({
+            View: FormView,
+            model: 'foo',
+            data: this.data,
+            res_id: 1,
+            viewOptions: { mode: 'edit' },
+            arch: '<form>' +
+                    '<sheet>' +
+                        '<notebook>' +
+                            '<page string="Page1"></page>' +
+                            '<page string="Page2">' +
+                                '<field name="o2m">' +
+                                    '<tree editable="bottom">' +
+                                        '<field name="titi"/>' +
+                                        '<field name="grosminet"/>' +
+                                    '</tree>' +
+                                '</field>' +
+                            '</page>' +
+                        '</notebook>' +
+                    '</sheet>' +
+                '</form>',
+        });
+
+        assert.isNotVisible(form.$('.o_field_one2many'),
+            "One2many field should be hidden");
+
+        await testUtils.dom.click(form.$('.nav-item:last-child .nav-link'));
+
+        assert.isVisible(form.$('.o_field_one2many'),
+            "One2many field should be visible");
+
+        let [titi, grosminet] = form.$('.tab-pane:last-child th');
+        assert.strictEqual(titi.style.width, "",
+            "width of small char should not be set yet");
+        assert.strictEqual(grosminet.style.width, "",
+            "width of large char should also not be set");
+
+        // Helper function that unfreezes all columns and asserts that the list has been
+        // frozen after a given callback.
+        const assertFreeze = async (callback, description) => {
+            const $o2m = form.$('.tab-pane:last-child .o_field_one2many');
+            $o2m.find('th').each((index, th) => {
+                th.style.width = null;
+            });
+            $o2m.find('table')[0].style.tableLayout = 'auto';
```

Comment:
I'm not a fan of this... This makes the test a bit too much artificial. Could you considerer spawning another form instance for each operation? You can do it in two lines (each time) if you define the params on the top. That way, you can get rid of assertFreeze.

Reply:
I'm not sure I fully understand, do you mean to create a brand new form view for each assertion ? And if so, how would it make assertFreeze pointless since one of its goals is to assert that the list is frozen ?

Reply:
The reason why you needed the assertFreeze helper is because you do more than one iteration (at the first one, the list is frozen so you need to unfreeze it for the next one). Am I right? If you do a single operation on each instance, you don't need to do this anymore (cf the other test)

Reply:
So, split the test into smaller ones where I assert one case at a time ?

Reply:
This is a way to do it, but I rather suggested to repeat those steps in a single test (such that you configure it only once):
 - create form view
 - do the operation (e.g. add a record)
 - assert
 - destroy